constraint/pkg/client/error_map.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// ErrorMap is a map from targets to the error the target returned.
+type ErrorMap map[string]error
+
+// Error implements error.
+//
+// Uses a pointer receiver to avoid potential errors.Is() bugs.
+func (e *ErrorMap) Error() string {
+	b := &strings.Builder{}
+
+	// Make printed error deterministic by sorting keys.
+	keys := make([]string, len(*e))
+	i := 0
+	for k := range *e {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		b.WriteString(fmt.Sprintf("%s: %s\n", k, (*e)[k]))
+	}
+	return b.String()
+}
+
+func (e *ErrorMap) Is(target error) bool {
```

Comment:
Acknowledged. It's possible as-is, but I wouldn't add it unless multiple callers need that functionality. Since ErrorMap is a non-struct non-interface type definition *(is there a shorter name for that?)* callers are able to do this themselves.